### PR TITLE
Feat: add Docker+wasm examples

### PR DESCRIPTION
**Pull request by juntao**
Added two multi-container examples to showcase the capability of Docker + Wasm to support applications with both Linux containers and Wasm containers running side by side.

Also added a new SVG icon to indicate samples that require Docker + wasm to build and run.

#306

### CODE REVIEW

File: wasmedge-mysql-nginx/README.md
Comment (dvdksn) [inline wasmedge-mysql-nginx/README.md:90]: Adding a new section/heading for how to work with the API using curl

```suggestion

### Using the API with `curl`

As an alternative to the web frontend, you can use `curl` to interact with the WasmEdge API directly (the `backend` service).

When the WasmEdge web service receives a GET request to the `/init` endpoint, it would initialize the database with the `orders` table.
```

File: wasmedge-mysql-nginx/README.md
Comment (dvdksn) [inline wasmedge-mysql-nginx/README.md:96]:
```suggestion
When the WasmEdge web service receives a POST request to the `/create_order` endpoint, it extracts the JSON data from the POST body and inserts an `Order` record into the database table.
```

File: wasmedge-mysql-nginx/README.md
Comment (dvdksn) [inline wasmedge-mysql-nginx/README.md:97]:
```suggestion
To insert multiple records, use the `/create_orders` endpoint and POST a JSON array of `Order` objects:
```

File: wasmedge-mysql-nginx/README.md
Comment (dvdksn) [inline wasmedge-mysql-nginx/README.md:103]:
```suggestion
When the WasmEdge web service receives a GET request to the `/orders` endpoint, it gets all rows from the `orders` table and return the result set in a JSON array in the HTTP response.
```

File: wasmedge-mysql-nginx/README.md
Comment (dvdksn) [inline wasmedge-mysql-nginx/README.md:109]:
```suggestion
When the WasmEdge web service receives a POST request to the `/update_order` endpoint, it extracts the JSON data from the POST body and update the `Order` record in the database table that matches the `order_id` in the input data.
```

File: wasmedge-mysql-nginx/README.md
Comment (dvdksn) [inline wasmedge-mysql-nginx/README.md:115]:
```suggestion
When the WasmEdge web service receives a GET request to the `/delete_order` endpoint, it deletes the row in the `orders` table that matches the `id` GET parameter.
```

**Review by dvdksn — COMMENTED** [review]
here are some update suggestions to the verbiage :) btw, can we use port 1024 or some other high port that is not port 80 for the web frontend?

**Comment by juntao** [comment]
Hi @dvdksn 

Thank you so much for your review and suggestions. I accepted the suggested changes. I also changed the web port to non-privileged port 8090 (default for the Nginx image). Thank you!

cheers
Michael

**Comment by juntao** [comment]
Hi @dvdksn @mikesir87 @rumpl 

Do we need another review in order to merge? 🙏 

Cheers
Michael

File: wasmedge-mysql-nginx/compose.yml
Comment (rumpl) [inline wasmedge-mysql-nginx/compose.yml:11]:
```suggestion
    platform: wasi/wasm3
```

File: wasmedge-kafka-mysql/compose.yml
Comment (rumpl) [inline wasmedge-kafka-mysql/compose.yml:23]:
```suggestion
    platform: wasi/wasm
```

File: wasmedge-mysql-nginx/.docker/docker-compose.yml
Comment (rumpl) [inline wasmedge-mysql-nginx/.docker/docker-compose.yml:11]:
```suggestion
    platform: wasi/wasm
```

File: wasmedge-mysql-nginx/backend/src/main.rs
Comment (rumpl) [inline wasmedge-mysql-nginx/backend/src/main.rs:256]: Can you remove this commented code?

File: wasmedge-mysql-nginx/compose.yml
Comment (juntao) [reply wasmedge-mysql-nginx/compose.yml:11]: I found that Docker Desktop 4.15 actually requires `wasi/wasm32`? Shall we wait until the next Docker Desktop release raise another PR to fix this? Thanks.

Comment (juntao) [reply wasmedge-mysql-nginx/compose.yml:11]: @rumpl

File: wasmedge-mysql-nginx/backend/src/main.rs
Comment (juntao) [reply wasmedge-mysql-nginx/backend/src/main.rs]: This is done in 

https://github.com/docker/awesome-compose/pull/309/commits/bbe1a6d1d1e467160e8d7609b6e7e44e3e2d82b1

Thanks.

File: wasmedge-mysql-nginx/compose.yml
Comment (mikesir87) [reply wasmedge-mysql-nginx/compose.yml:11]: @juntao - How did you see that Docker Desktop requires `wasi/wasm32`? I've been using `wasi/wasm` in many of my PoCs and haven't run into issues. It's possible you need to rebuild the images after making the change, as you won't have a matching image with the right platform.

Comment (juntao) [reply wasmedge-mysql-nginx/compose.yml]: My apologies. My Docker install had other problems before. I can confirm that `wasi/wasm` works on Docker Desktop 4.15. I have updated the code to reflect this. Thanks.

https://github.com/docker/awesome-compose/pull/309/commits/574e9807c6a12bf9bcd3b62deddd214ce4b11761

**Comment by juntao** [comment]
@rumpl and @mikesir87 

Happy new year! Can you review the changes? Thanks!

cheers
Michael

File: README.md
Comment (mikesir87) [inline README.md:23]:
```suggestion
<a href="https://docs.docker.com/desktop/wasm/"><img src="icon_wasm.svg" alt="Docker + wasm" height="30" align="top"/></a> Icon indicates Sample is compatible with [Docker+Wasm](https://docs.docker.com/desktop/wasm/).
```

**Review by mikesir87 — APPROVED** [review]
One small tweak, but LGTM

**Comment by dvdksn** [comment]
:tada:

**Comment by juntao** [comment]
Thank you! And happy new year! 🎈

**Pull request merged**
